File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

a = int(input())
a_list = list(map(int,input().split()))

b = int(input())
b_list = list(map(int,input().split()))

a_list.sort()

def binary_search(target):
    left, right = 0, a-1
    
    while left <= right:
        mid = (left + right) // 2
        if target > a_list[mid]:
            left = mid + 1
        elif target < a_list[mid]:
            right = mid - 1
        else: 
            return 1
        
    return 0
    
for b1 in b_list:
    print(binary_search(b1))

File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.*;
import java.util.*;

public class Main {
	static int n;
	static int m;

	public static void main(String[] args) throws Exception {
//		System.setIn(new FileInputStream("res/input.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		n = Integer.parseInt(br.readLine());
		int[] a = new int[n];
		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++) {
			a[i] = Integer.parseInt(st.nextToken());
		}
		Arrays.sort(a);
		m = Integer.parseInt(br.readLine());
		st = new StringTokenizer(br.readLine());
		for (int i = 0; i < m; i++) {
			int x = Integer.parseInt(st.nextToken());
			int l = 0;
			int r = n - 1;
			int ans = 0;
			while (l <= r) {
				int mid = (l + r) / 2;
				if (a[mid] < x) {
					l = mid + 1;
				} else if (a[mid] > x) {
					r = mid - 1;
				} else {
					ans = 1;
					break;
				}
			}
			System.out.println(ans);
		}
	}
}